File: 2021-JS/Day-2/day2.js
const fs = require('fs');

let input = String(fs.readFileSync('D:\\coding\\Personal\\AdventOfCode\\2021\\Day-2\\input.txt', 'utf-8'));
let lines = input.split('\n');

Part1(lines);
Part2(lines);

function Part1(input){
    if(Array.isArray(input)){
        let depth = 0;
        let position = 0;
        for(let i = 0; i < input.length; i++){
            let line = input[i].split(' ');
            if(line[0] == "down"){
                depth += Number(line[1]);
            }
            else if(line[0] == "up"){
                depth -= Number(line[1]);
            }
            else if(line[0] == "forward"){
                position += Number(line[1]);
            }
        }
        console.log(position * depth);
    }
}
function Part2(input){
    if(Array.isArray(input)){
        let aim = 0;
        let depth = 0;
        let position = 0;
        for(let i = 0; i < input.length; i++){
            let line = input[i].split(' ');
            if(line[0] == "down"){
                aim += Number(line[1]);
            }
            else if(line[0] == "up"){
                aim -= Number(line[1]);
            }
            else if(line[0] == "forward"){
                position += Number(line[1]);
                depth += aim * Number(line[1]);
            }
        }

        console.log(position * depth);
    }
}

File: 2019-Python/Day-2/day2.py
def Part1(opcode):
    index = 0

    #position 1 is replaced with 12 and position 2 is replaced with 2
    opcode[1] = 12
    opcode[2] = 2
    while True:
        print(opcode[index])
        if index >= len(opcode):
            break
        elif int(opcode[index]) == 1:
            opcode[int(opcode[index + 3])] = int(opcode[int(opcode[index + 1])]) + int(opcode[int(opcode[index + 2])])
            index +=4
        elif int(opcode[index]) == 2:
            opcode[int(opcode[index + 3])] = int(opcode[int(opcode[index + 1])]) * int(opcode[int(opcode[index + 2])])
            index +=4
        elif int(opcode[index]) == 99:
            break

    print(opcode)
    print(opcode[0])
    return 0

with open('input.txt', 'r') as file:
    content = file.read()

opcode = content.split(',')

Part1(opcode)